packages/devextreme/js/__internal/ui/calendar/m_calendar.ts

```diff
@@ -251,116 +261,134 @@ class Calendar<
 
         this._moveToClosestAvailableDate(date);
       },
-      pageUp(e): void {
+      pageUp(e: KeyboardEvent): void {
         e.preventDefault();
         this._waitRenderView(-1 * this._getRtlCorrection());
       },
-      pageDown(e): void {
+      pageDown(e: KeyboardEvent): void {
         e.preventDefault();
         this._waitRenderView(1 * this._getRtlCorrection());
       },
-      tab() {},
+      tab(): void {},
       enter: this._enterKeyHandler,
     };
   }
 
-  _enterKeyHandler(e): void {
+  _enterKeyHandler(e: KeyboardEvent): void {
+    const { currentDate = new Date() } = this.option();
     if (!this._isMaxZoomLevel()) {
       this._navigateDown();
-    } else if (!this._view.isDateDisabled(this.option('currentDate'))) {
-      const value = this._updateTimeComponent(this.option('currentDate'));
+    } else if (!this._view.isDateDisabled(currentDate)) {
+      const value = this._updateTimeComponent(currentDate);
       this._selectionStrategy.selectValue(value, e);
     }
   }
 
-  _getSerializationFormat(optionName) {
-    const value = this.option(optionName || 'value');
+  _getSerializationFormat(optionName = 'value'): string | undefined | null {
+    const value = this.option(optionName);
```

Comment:
In this notation typescript can't determine value type, it shows
` const value: TProperties extends CalendarProperties = CalendarProperties`

Use the following notation 
```
  _getSerializationFormat(optionName: 'value' | 'min' | 'max' = 'value'): string | undefined | null {
    const { [optionName]: value } = this.option();
    ...
 ```

Reply:
https://github.com/DevExpress/DevExtreme/commit/ec2f610d374bc1eb469a6ab73af78a85080ffa5d